diff --git a/colorizer.module b/colorizer.module
index 192f9ba..2d98083 100644
--- a/colorizer.module
+++ b/colorizer.module
@@ -255,6 +255,8 @@ function colorizer_update_stylesheet($theme, $instance, $palette) {
     $target_styles = _colorizer_rewrite_stylesheet($palette, $source_styles);
     $target_file = $target_path . $id . '.css';
     _colorizer_save_stylesheet($target_file, $target_styles);
+    // Keep track of the CSS file that's currently being used.
+    variable_set('colorizer_current_sylesheet', $target_file);
   }
 
   if (!empty($target_file)) {
@@ -287,11 +289,39 @@ function _colorizer_save_stylesheet($file, $style) {
 }
 
 /**
- * Removes css files from colorizer files cache
- * They will get recreated in hook_init on demand
+ * Removes unused css files from colorizer files cache.
  */
 function colorizer_clearcache() {
-  file_unmanaged_delete_recursive('public://colorizer');
+
+  $colorizer_path = 'public://colorizer';
+  // Get the name of the latest CSS file.
+  $current_stylesheet = variable_get('colorizer_current_sylesheet', '');
+
+  // check if the colorizer directory exists.
+  if (is_dir($colorizer_path)) {
+
+    $dir = dir($colorizer_path);
+    // Look through the directory content.
+    while (($entry = $dir->read()) !== FALSE) {
+
+      // Skip '.' and '..'.
+      if ($entry == '.' || $entry == '..') {
+        continue;
+      }
+
+      $entry_path = $colorizer_path . '/' . $entry;
+
+      // Delete all the CSS files that are not currently being used.
+      // The latest file cannot be deleted in order to avoid having issues when the Poor Man's Cron runs:
+      // - The hook_preprocess_html() is called and the CSS file is added to the page.
+      // - The Poor Man's Cron runs as part of the page requests and all unused files are deleted.
+      // - Since the latest file was not deleted then the CSS that was added before cron is still available.
+      if ($entry_path != $current_stylesheet) {
+        file_unmanaged_delete($entry_path);
+      }
+    }
+    $dir->close();
+  }
 }
 
 /**
