diff --git a/template.php b/template.php
index 2a93f7c..5540f90 100644
--- a/template.php
+++ b/template.php
@@ -16,10 +16,12 @@ foreach (glob($includes_path) as $filename) {
 function radix_preprocess_html(&$variables) {
   global $base_url;
 
-//  // Add Bootstrap JS from CDN if bootstrap library is not installed.
+  // Adding Bootstrap JS from CDN causes cross-domain javascript source
+  // file inclusion security errors. So load a local copy from nuboot_radix.
+  // Using bootstrap_library breaks recline functionality.
   if (!module_exists('bootstrap_library')) {
-    $base = parse_url($base_url);
-    $url = $base['scheme'] . '://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js';
+    $theme = drupal_get_path('theme', 'nuboot_radix');
+    $url = $base_url . '/' . $theme . '/assets/js/bootstrap.min.js';
     $jquery_ui_library = drupal_get_library('system', 'ui');
     $jquery_ui_js = reset($jquery_ui_library['js']);
     drupal_add_js($url, array(

